### PR TITLE
Moving to most recent nav-common library

### PR DESCRIPTION
**Pull request by dragutin-nav**

### CODE REVIEW

File: src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public AktorClient aktorClient(EnvironmentProperties properties, SystemUserToken
                 properties.getAktorregisterUrl(), APPLICATION_NAME, systemUserTokenProvider::getSystemUserToken
         );
 
-        return new AktorClient(new CachedAktorOppslagClient(aktorOppslagClient), new CachedAktorregisterClient(aktorregisterClient), unleashService);
+        return new AktorClient(new CachedAktorOppslagClient(aktorOppslagClient), aktorregisterClient, unleashService);
```
Comment (dragutin-nav) [inline src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java:44]: @stianStensli maybe you can help me here...I couldnt find class for caching aktorregiste

File: src/main/java/no/nav/pto/veilarbportefolje/config/ApplicationConfig.java [hunk 2]
```diff
@@ -34,11 +33,6 @@ public MetricsReporter elasticMetricsReporter(ElasticIndexer elasticIndexer) {
         return new MetricsReporter(elasticIndexer);
     }
 
-    @Bean
-    public UnleashService unleashService() {
-        return new UnleashService(UnleashServiceConfig.resolveFromEnvironment());
-    }
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/config/ApplicationConfig.java:40]: Nice to use a service 👍

File: src/main/java/no/nav/pto/veilarbportefolje/kafka/KafkaConfigCommon.java [hunk 3]
```diff
@@ -41,52 +37,43 @@
...
-    @Bean
-    public LockProvider lockProvider(JdbcTemplate jdbcTemplate) {
-        return new JdbcTemplateLockProvider(jdbcTemplate);
-    }
-
-    @Bean
-    public KafkaConsumerRepository kafkaConsumerRepository(@Qualifier("Postgres") DataSource dataSource) {
-        return new PostgresConsumerRepository(dataSource);
-    }
+    private final KafkaConsumerClient consumerClient;
```
Comment (dragutin-nav) [inline src/main/java/no/nav/pto/veilarbportefolje/kafka/KafkaConfigCommon.java:40]: this logic is from veilarboppfolging..

File: src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java [hunk 4]
```diff
@@ -1,17 +1,18 @@
 package no.nav.pto.veilarbportefolje.client;
 
-import no.nav.common.auth.context.AuthContextHolder;
+import no.nav.common.auth.context.AuthContextHolderThreadLocal;
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:3]: AuthContextHolderThreadLocal I need to check why it is called ThreadLocal 🤔  does this mean that it is not meant for prod?

File: src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java (hunk 1 above)
Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java:44]: I will ask Alexander 👍

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java:44]: Alexander seems to want us to switch to only using CachedAktorOppslagClient, but that one does not work well with real users in q1....

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java:44]: will it be possible to do something like this?
CachedAktorOppslagClient(new AktorregisterHttpClient())

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/ClientConfig.java:44]: Another option: use ToggledAktorOppslagClient

File: src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java (hunk 4 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:3]: I dont think its related with env...just its named like that

File: src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
-            return AuthContextHolder.getIdTokenString()
-                    .map(RestClientUtils::createBearerToken)
-                    .orElseThrow(IllegalStateException::new);
-        }
+    public static String authHeaderMedSystemBruker() {
+        return AuthContextHolderThreadLocal
+                .instance().getIdTokenString()
+                .map(RestClientUtils::createBearerToken)
+                .orElseThrow(IllegalStateException::new);
+    }
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:12]: Comment from Alexander: Nja, det vil fungere, men tanken bak AuthContextHolderThreadLocal er at den skal lages som en bønne og injectes inn de stedene det trengs. Grunnen til en bønne fungerer bedre er fordi da har du mulighet til å mocke ut AuthContextHolderThreadLocal når man lager tester

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:12]: @Bean
    public AuthContextHolder authContextHolder() {
        return AuthContextHolderThreadLocal.instance();
    }

File: src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java (hunk 4 above)
Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:3]: you are correct 👍  but I added a comment on the same issue after a talk with alexander

File: src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java (hunk 5 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:12]: Yep, I will try to improve that...although its not everywhere possible to inject bean

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/client/RestClientUtils.java:12]: true and he said that i should work so there is no worry :)

**Comment by dragutin-nav** [comment]
@stianStensli can you pls check again...I made some changes...

**Review by stianStensli — APPROVED** [review]

**Pull request merged**